storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManager.java

```diff
@@ -1120,6 +1111,16 @@ void resetLagStats() {
             brokerTopicStats.recordRemoteCopyLagSegments(topic, partition, 0);
         }
 
+        private void recordLagStats(UnifiedLog log) {
+            try {
+                long bytesLag = log.onlyLocalLogSegmentsSize() - log.activeSegment().size();
+                long segmentsLag = log.onlyLocalLogSegmentsCount() - 1;
+                recordLagStats(bytesLag, segmentsLag);
+            } catch (Exception e) {
```

Comment:
This change can emit negative values when there are no candidate segments to upload and the `highestOffsetInRemoteStorage` is higher than the base-offset of the active segment. This can happen after leader change:

We can update this to ensure that non-negative values are not emitted: 
```
long bytesLag = Math.max(0, log.onlyLocalLogSegmentsSize() - log.activeSegment().size());
long segmentsLag = Math.max(0, log.onlyLocalLogSegmentsCount() - 1);
```

This issue wasn't there previously as the metric gets reported only when the candidate segments are available. Could you cover this PR with unit tests?